### DISCUSSION TITLE
top level 'await' expressions in the REPL

### DISCUSSION THREAD

**Discussion by andy-98point6** [TypeStrong/ts-node#1299] [Q&A] [answered]
I want to write something like this:

```
async function sendRequest() {...}
const res = await sendRequest(...) // HTTP get
console.log(res)
```

But I get this error:
`Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext' or 'system', and the 'target' option is set to 'es2017' or higher.`

I am invoking the repl with `npx ts-node`.  How can I find how to set the module option to system?

**Comment by cspotcode** [comment, marked as answer]
Check out these sections of the readme:
https://github.com/TypeStrong/ts-node#configuration-options
https://github.com/TypeStrong/ts-node#cli-and-programmatic-options
https://github.com/TypeStrong/ts-node#loading-tsconfigjson

**Reply by cspotcode** [reply]
However, I don't think top-level await works in the REPL.

`node` itself has this gated behind an experimental flag (read: does not follow semver, may break or change)
https://nodejs.org/dist/latest-v16.x/docs/api/repl.html#repl_await_keyword

I've looked a bit at their code to make this work, and it's hacky.  I don't think they've exposed this functionality through their REPL API, which we would probably need in order to support the same in `ts-node`.
